tests/unit/test_record.py

```diff
@@ -216,3 +227,60 @@ def test_func(a: int, b: str, c: Optional[str] = None) -> str:
     assert recorder._records_by_type["TestAutoRecord"][-1].params.a == 123
     assert recorder._records_by_type["TestAutoRecord"][-1].params.b == "abc"
     assert recorder._records_by_type["TestAutoRecord"][-1].result.return_val == "123abc"
+
+
+def test_recorded_function_with_override() -> None:
+    os.environ["DBT_RECORDER_MODE"] = "Record"
+    recorder = Recorder(RecorderMode.RECORD, None)
+    set_invocation_context({})
+    get_invocation_context().recorder = recorder
+
+    @supports_replay
+    class Recordable:
+        @auto_record_function("TestAuto")
+        def test_func(self, a: int) -> int:
+            return 2 * a
+
+    class RecordableSubclass(Recordable):
+        def test_func(self, a: int) -> int:
+            return 3 * a
+
+    rs = RecordableSubclass()
+
+    rs.test_func(1)
+
+    assert recorder._records_by_type["TestAutoRecord"][-1].params.a == 1
+    assert recorder._records_by_type["TestAutoRecord"][-1].result.return_val == 3
+
+
+class CustomType:
+    def __init__(self, n: int):
+        self.value = n
+
+
+class CustomSerializationStrategy(SerializationStrategy):
+    def serialize(self, obj: CustomType) -> int:
+        return obj.value
+
+    def deserialize(self, value: int) -> CustomType:
+        return CustomType(value)
+
+
+def test_recorded_with_custom_serializer() -> None:
+    os.environ["DBT_RECORDER_MODE"] = "Record"
+    recorder = Recorder(RecorderMode.RECORD, None)
+    set_invocation_context({})
+    get_invocation_context().recorder = recorder
+
+    recorder.register_serialization_strategy(CustomType, CustomSerializationStrategy())
```

Comment:
This is great! I guess we can just maintain a file of type: serialization  later on for all of the types we encounter?